woocommerce/woocommerce-ios
# [Orders] Ability to filter by product

Part of: #12059 


## Description

Allows merchants to view orders by filtering them using a product. 

- Adds a new row to select the product for filtering orders
- The selected product info is saved to settings and restored upon app relaunch. 


## Testing instructions

- Create a store
- Create a few products and orders
- Login into the app
- Navigate to the Orders tab
- Tap "Filter" -> "Product" 
- From the list of products select one
- Tap "Show Orders"
- The order list screen should show only the orders related to the selected order now.
- Tap "Filter" again and validate that the previously selected product name is visible in the filter.
- Tap "Clear all" to clear the filters. 
- Applying the filter should start showing all the orders in the order list screen.

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/524475/1f87f218-58fe-4fae-920b-7fb5d47def1f



---
